src/toxcore/dht/server/mod.rs

```diff
@@ -530,10 +549,20 @@ impl Server {
             &payload
         ));
         self.send_to_direct(node.saddr, ping_req)
+            .then(|r| {
```

Comment:
I don't like this block added multiple times to sending packet methods. A fortiori, we didn't filter out timeout error before. So I'd add this error to the return type: `Error = TimeoutError<mpsc::SendError<(Packet, SocketAddr)>>`.